docs/docs/crafty.md

```diff
@@ -3,4 +3,184 @@ id: crafty
+
+In [Crafty](https://crafty.zeppelin.solutions), users can create new ERC20 tokens, with a few added goodies (following [EIP-1046](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1046.md)): a picture, and a short description. This allows for each token to have that extra bit of personality that makes something unique. But that's not all: the only way to get these tokens is by crafting them, which requires ingredients (any ERC20!) to be spent. And since the craftable tokens themselves can also be used as ingredients, the posibilities are endless!
+
+## Project setup
+
+This guide will show you the process of making the contracts in the Crafty game upgradeable by using ZeppelinOS, which we'll install globally to have it available on all our projects.
+
+```sh
+npm install --global zos
+```
+
+All snippets here were extracted from the public [Crafty repository](https://github.com/zeppelinos/crafty).
+
+## Contracts
+
+Crafty works with only two contracts:
+* [`Crafty`](https://github.com/zeppelinos/crafty/blob/master/contracts/Crafty.sol) itself, containing the game logic (how new recipes are added, how crafting takes place, etc.) plus some role-based access control ([RBAC](https://github.com/OpenZeppelin/openzeppelin-solidity/blob/zos-release/contracts/ownership/rbac/RBAC.sol), powered by [OpenZeppelin](https://openzeppelin.org/)), for admin tasks, such as deletion of tokens.
+* The [`CraftableToken`](https://github.com/zeppelinos/crafty/blob/master/contracts/CraftableToken.sol)s, which are a simple extension of a mintable ERC20, holding token creator data, a URI for metadata, and a list of required ingredients, along with the amounts for each.
+
+To better see how these two contracts interact, take a look at `Crafty`'s `craft` method, where a user provides the address of a `CraftableToken` to craft, and `Crafty` makes sure the required ingredients are consumed before minting it.
+
+```
+function craft(CraftableToken _craftable) public {
+  address player = msg.sender;
+
+  uint256 totalSteps = _craftable.getTotalRecipeSteps();
+  for (uint i = 0; i < totalSteps; ++i) {
+    ERC20 ingredient;
+    uint256 amountNeeded;
+    (ingredient, amountNeeded) = _craftable.getRecipeStep(i);
+
+    ingredient.transferFrom(player, address(this), amountNeeded);
+  }
+
+  // Issue the crafted token - for ingredient-less tokens, a higher number of
+  // tokens are minted.
+  uint256 tokensToMint = (totalSteps == 0) ? 100 : 1;
+  _craftable.mint(player, tokensToMint);
+}
+```
+
+## Developing with ZeppelinOS
+
+So, how would we go about integrating ZeppelinOS to such a project? The process is remarkably simple: we'll cover it step by step. But before we get started, don't forget to initialize your project with ZeppelinOS:
+
+```sh
```

Comment:
Before this, shouldn't we talk about earlier bootstrapping? i.e. npm init, truffle init, etc

Reply:
I discussed this with @fiiiu beforehand: I explicitly avoided those steps since we already discussed them everywhere else to focus on the core of the guide, which is supposed to be one of the last ones.

Reply:
Maybe a compromise would be to add a block of shell code with the (3 or 4, I expect) lines needed for bootstrapping, and a link to the setup guide as an explanation for why this is such. 

More generally: put the needed commands, save the explanations of setup, linking to prior guides (setup, building, using, basil).